judge/views/problem.py

```diff
@@ -200,9 +200,69 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        # The vote this user has already cast on this problem.
+        if context['can_vote']:
+            try:
+                context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                context['vote'] = None
+        else:
+            context['vote'] = None
+
+        all_votes = list(self.object.problem_points_votes.order_by('points').values_list('points', flat=True))
+        context['has_votes'] = len(all_votes) > 0
+
+        # If the user is not currently in contest.
+        if not user.is_authenticated or user.profile.current_contest is None:
+            context['all_votes'] = all_votes
+
+        context['max_possible_vote'] = settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE
+        context['min_possible_vote'] = max(settings.DMOJ_PROBLEM_MIN_USER_POINTS_VOTE,
+                                           settings.DMOJ_PROBLEM_MIN_PROBLEM_POINTS)
         return context
 
 
+class DeleteVote(ProblemMixin, SingleObjectMixin, View):
```

Comment:
Aren't the rest of these views called something like `ProblemXXXX`? Make this `DeleteProblemVote`, and fix the one below too.